File: 디자인패턴/java 디자인 패턴/Mediator/CoolAircon.java
public class CoolAircon extends Participant {
    private boolean bOff = false;
    
    public CoolAircon(Mediator mediator) {
        super(mediator);
    }

    public void On() {
        bOff = true;
        mediator.participantChange(this);
    }

    public void Off() {
        bOff = false;
        mediator.participantChange(this);
    }

    public boolean isRunning() {
        return bOff;
    }

    @Override
    public String toString() {
        if(bOff) return "# 에어컨: 켜짐";
        else return "# 에어컨: 꺼짐";
    }
}

File: 디자인패턴/Python 디자인 패턴/Mediator/CoolAircon.py
from Participant import*

class CoolAircon(Participant):
    def __init__(this, mediator: Mediator) -> None:
        super().__init__(mediator)
        this.bAircon = False
    
    def On(this):
        this.bAircon = True
        this.mediator.participantChange(this)
    
    def Off(this):
        this.bAircon = False
        this.mediator.participantChange(this)
    
    def isAircon(this):
        return this.bAircon
    
    def strPrint(this):
        if(this.bAircon):return "# 에어컨 켜짐"
        return "# 에어컨 꺼짐"
    pass